webview/App.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { useEffect, useRef, useState } from 'react'
+import { throttle } from './utils'
+
+declare var acquireVsCodeApi: any
+
+export default function AliveREPL() {
+    const [vscodeApi, setVscodeApi] = useState(null)
+    const scrollReplRef = useRef(null)
+
+    const [packageName, setPackageName] = useState('')
+    const [inputText, setInputText] = useState('')
+    const [outputItems, setOutputItems] = useState([])
+
+    const inputRef = useRef(null)
+    const replOutputRef = useRef(null)
+    const calledOnce = React.useRef(false)
+
+    const messageHandler = (event) => {
+        const data = event.data
+
+        switch (data.type) {
+            case 'setInput':
+                setInputText(data.text)
+                break
+            case 'clearInput':
+                setInputText('')
+                break
+            case 'setOutput':
+                setOutputItems(data.items)
+                scrollReplRef.current()
+                break
+            case 'appendOutput':
+                setOutputItems(prevOutputItems => [...prevOutputItems, data.obj])
+                scrollReplRef.current()
+                break
+            case 'clear':
+                setOutputItems([])
+                break;
+            case 'setPackage':
+                setPackageName(data.name)
+                inputRef.current?.focus()
+                break
+            default:
+                break
+        }
+    }
+
+    useEffect(() => {
+        if (calledOnce.current) return
+        calledOnce.current = true
+
+        // VSCode instance can be acquired only once
+        setVscodeApi(acquireVsCodeApi())
+
+        // Scroll the output view with a delay
+        scrollReplRef.current = throttle(() => scrollReplView(), 80)
+    }, [])
+
+    useEffect(() => {
+        window.addEventListener('message', messageHandler)
+        return () => window.removeEventListener('message', messageHandler)
+    }, [messageHandler])
```

Comment:
Need to set up the linting so it'll complain about these lists. If I remove `messageHandler`, it doesn't do anything.  The biggest issue with React is how easily it can go into a render loop because of the hooks, so it needs to check that the dependency lists are accurate.